storage/localstore/mode_get.go

```diff
@@ -96,6 +97,15 @@ func (db *DB) get(mode chunk.ModeGet, addr chunk.Address) (out shed.Item, err er
 			}
 		}()
 
+	case chunk.ModeGetPin:
+		pinChunk, err := db.pinIndex.Get(item)
+		if err != nil {
+			return out, err
+		}
+		pinChunk.Data = make([]byte, 8)
+		binary.BigEndian.PutUint64(pinChunk.Data[:8], pinChunk.PinCounter)
```

Comment:
Encoding the pin counter in the Data field? Why do not we provide PinCounter on the Chunk type (interface) instead, instead cryptically encoding the integer into the data field? NewChunk constructor can accept additional integer from the item (`out` variable) in DB.Get. This way there will be no need for decoding pin counter in other places and deal with unexpected chunk data in the rest of the codebase. Chunk data should be just about the data, and not about encoding various things from it, that is why we have data structures, too, not just byteslices to work with.

Reply:
This was one thing that was itching me too. I was too scared to change chunk as it is sort of very centre to so many things. Anyway.. changed  Chunk interface now to store and give back pinCounter too.

Reply:
Thank you.